### PR TITLE
JBPM-7358 Allow filtering process instances by SLA status

### PR DESCRIPTION
**Pull request by bxf12315**

### CODE REVIEW

**Comment by bxf12315** [comment]
The PR dependent on https://github.com/kiegroup/appformer/pull/650.

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/basic/BasicFiltersView.java [hunk 1]
```diff
@@ -61,5 +61,12 @@ void addDataSetSelectFilter(String label,
                                 String valueColumnId,
                                 Consumer<ActiveFilterItem<String>> callback);
 
+    void addDataSetSelectFilter(String label,
```
Comment (cristianonicolai) [inline jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/basic/BasicFiltersView.java]: why not use _addSelectFilter_? Seems more appropriate for a static list of values like this. DataSetSelectFilter is based on dynamic data from db.

**Comment by bxf12315** [comment]
@cristianonicolai I think you are right, I will change it.

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/basic/BasicFiltersView.java (hunk 1 above)
Comment (bxf12315) [reply jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/basic/BasicFiltersView.java]: updated, Please review again.

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java [hunk 2]
```diff
@@ -175,11 +177,24 @@ protected void addProcessIdFilter() {
                                     f -> addSearchFilter(f, equalsTo(COLUMN_PROCESS_ID, f.getValue())));
     }
 
+    protected void addSLAComplianceFilter() {
+        Map<String, String> aliasMap = new HashMap<String, String>();
+        aliasMap.put("0", constants.SlaNA());
+        aliasMap.put("1", constants.SlaPending());
+        aliasMap.put("2", constants.SlaMet());
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: Probably better to use the static references we already have:
```suggestion
        aliasMap.put(String.valueOf(ProcessInstance.SLA_MET), constants.SlaMet());
```

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java [hunk 3]
```diff
@@ -175,11 +177,24 @@ protected void addProcessIdFilter() {
...
+        aliasMap.put("3", constants.SlaAborted());
+        aliasMap.put("4", constants.SlaViolated());
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: 4 = SLA_ABORTED

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java [hunk 4]
```diff
@@ -175,11 +177,24 @@ protected void addProcessIdFilter() {
                                     f -> addSearchFilter(f, equalsTo(COLUMN_PROCESS_ID, f.getValue())));
     }
 
+    protected void addSLAComplianceFilter() {
+        Map<String, String> aliasMap = new HashMap<String, String>();
+        aliasMap.put("0", constants.SlaNA());
+        aliasMap.put("1", constants.SlaPending());
+        aliasMap.put("2", constants.SlaMet());
+        aliasMap.put("3", constants.SlaAborted());
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: 3 = SLA_VIOLATED

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java [hunk 5]
```diff
@@ -175,11 +177,24 @@ protected void addProcessIdFilter() {
                                     f -> addSearchFilter(f, equalsTo(COLUMN_PROCESS_ID, f.getValue())));
     }
 
+    protected void addSLAComplianceFilter() {
+        Map<String, String> aliasMap = new HashMap<String, String>();
+        aliasMap.put("0", constants.SlaNA());
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: Please also add options in alphabetical order, makes easier to select in the UI.

**Comment by bxf12315** [comment]
@cristianonicolai updated, Please review it.

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java [hunk 6]
```diff
@@ -191,6 +193,19 @@ protected void addDeploymentIdFilter() {
                                     f -> addSearchFilter(f, equalsTo(COLUMN_EXTERNAL_ID, f.getValue())));
     }
 
+    protected void addSLAComplianceFilter() {
+        Map<String, String> aliasMap = new HashMap<String, String>();
+        aliasMap.put(String.valueOf(ProcessInstance.SLA_ABORTED), constants.SlaAborted());
+        aliasMap.put(String.valueOf(ProcessInstance.SLA_MET), constants.SlaMet());
+        aliasMap.put(String.valueOf(ProcessInstance.SLA_NA), constants.SlaNA());
+        aliasMap.put(String.valueOf(ProcessInstance.SLA_PENDING), constants.SlaPending());
+        aliasMap.put(String.valueOf(ProcessInstance.SLA_MET), constants.SlaMet());
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: this is duplicate row.

Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: updated.

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Pull request merged**